common/lib/xmodule/xmodule/tests/test_html_module.py

```diff
@@ -24,6 +24,39 @@ def instantiate_descriptor(**field_data):
     )
 
 
+class HtmlModuleCourseApiTestCase(unittest.TestCase):
+    """
+    Ensure that student_view_data can handle likely input,
+    and doesn't modify the HTML in any way.
+    """
+    descriptor = Mock()
+
+    def prep_module(self, html):
+        field_data = DictFieldData({'data': html})
+        module_system = get_test_system()
+        return HtmlModule(self.descriptor, module_system, field_data, Mock())
+
+    def test_default(self):
+        sample_html = '<h1>Some content</h1>'
+        module = self.prep_module(sample_html)
+        self.assertEqual(module.student_view_data(), {'html': sample_html})
+
+    def test_blank_input(self):
+        sample_html = ''
+        module = self.prep_module(sample_html)
+        self.assertEqual(module.student_view_data(), {'html': sample_html})
+
+    def test_none(self):
+        sample_html = None
+        module = self.prep_module(sample_html)
+        self.assertEqual(module.student_view_data(), {'html': sample_html})
+
+    def test_invalid_html(self):
+        sample_html = '<h1>Some content</h'
+        module = self.prep_module(sample_html)
+        self.assertEqual(module.student_view_data(), {'html': sample_html})
```

Comment:
@bradmerlin This collection of tests can be collapsed to a single so called parameterized test (aka data-driven test).

I believe edx-platform already has library that provides support for parameterized tests installed - `ddt`.

Please refer to `ddt` docs for details and some other test modules for examples, but generally it would look something like this:

```
@ddt.ddt
class MyTest(Testcase):

    @ddt.data(1, 2, 3, 4)
    def test_with_single_parameter()self, parameter):
        pass

    @ddt.unpack
    @ddt.data(
        ("Parameter Group 1", 1),
        ("Parameter Group 2", 2),
    )
    def test_with_multiple_parameters(self, param1, param2):
        pass
```

Also, might be good to demonstrate/document/ensure that student_view_html is not designed to prevent XSS (i.e. script tags are not escaped), can carry images and is not limited to certain length of text. This later one might be a bit puzzling and/or tricky to implement - so let's rephrase it to "can handle very long strings".

Hint: constructing a very long string can be automated: [complex](https://stackoverflow.com/questions/2257441/random-string-generation-with-upper-case-letters-and-digits-in-python) or simple `'mystring' * 95`.